doc/source/data/transforming-data.rst

```diff
@@ -218,6 +219,15 @@ NumPy functions and model inference. However, if your batch size is too large, y
 program might run out of memory. If you encounter an out-of-memory error, decrease your
 ``batch_size``.
 
+.. _ordering_of_rows:
+
+Ordering of rows
+================
+When applying operations on data such as :meth:`~ray.data.Dataset.map` or :meth:`~ray.data.Dataset.map_batches`,
+the order of :ref:`blocks <dataset_concept>` is not preserved by default. If the order of blocks needs to be preserved,
+you can use :meth:`~ray.data.Dataset.sort` method, or set :attr:`ray.data.ExecutionOptions.preserve_order` to `True`.
```

Comment:
Could you add a quick code snippet?

Reply:
Not sure what do you mean. Some code snippets from https://github.com/ray-project/ray/issues/50890 ? Which one would you recommend?

Reply:
added below - thanks!